selfservice/form/html_form.go

```diff
@@ -84,20 +86,21 @@ func NewHTMLFormFromRequestBody(r *http.Request, action string, compiler decoder
 // NewHTMLFormFromJSON creates a HTML form based on the provided JSON struct.
 func NewHTMLFormFromJSON(action string, raw json.RawMessage, prefix string) *HTMLForm {
 	c := NewHTMLForm(action)
-	for k, v := range jsonx.Flatten(raw) {
-		if prefix != "" {
-			k = prefix + "." + k
-		}
-		c.SetValue(k, v)
-	}
+	c.SetValuesFromJSON(raw, prefix)
 
 	return c
 }
 
 // NewHTMLFormFromJSONSchema creates a new HTMLForm and populates the fields
 // using the provided JSON Schema.
 func NewHTMLFormFromJSONSchema(action, jsonSchemaRef, prefix string) (*HTMLForm, error) {
-	paths, err := jsonschemax.ListPaths(jsonSchemaRef, nil)
+	compiler := jsonschema.NewCompiler()
+	extensionRunner, err := schema.NewExtensionRunner(schema.ExtensionRunnerIdentityMetaSchema)
```

Comment:
I think it would make more sense if you would be able to pass a compiler as an argument to this function.